main.go

```diff
@@ -25,19 +36,53 @@ func main() {
 		log.Fatalf("API_KEY environment variable not set")
 	}
 
+	urlString := os.Getenv("BaseURL")
+	if urlString == "" {
+		log.Fatalf("BaseURL environment variable not set")
+	}
+
+	dbURI := os.Getenv("dbURI")
+	if dbURI == "" {
+		log.Fatalf("dbURI environment variable not set")
+	}
+
+	// Copy pasted from mongoDB website- setting up the db connection
+	// Use the SetServerAPIOptions() method to set the Stable API version to 1
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
+	opts := options.Client().ApplyURI(dbURI).SetServerAPIOptions(serverAPI)
+	// Create a new client and connect to the server
+	client, err := mongo.Connect(context.TODO(), opts)
+	if err != nil {
+		log.Fatalf("Counld'nt create a client to connect to the db")
+	}
+	// To disconnect the client from the db when the main ends
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			log.Fatalf("Couldn't disconnect the db connection ")
+		}
+	}()
+	// Send a ping to confirm a successful connection
+	var result bson.M
+	if err := client.Database("admin").RunCommand(context.TODO(), bson.D{{"ping", 1}}).Decode(&result); err != nil {
+		log.Fatalf("Couldn't connect to the database")
+	}
+	// log.Println(result)
+	log.Println("Pinged your deployment. You successfully connected to MongoDB!")
+
```

Comment:
Testing the connection getting the url for connecting can be a separate function ?
would make the main file more clearer.

Reply:
Hmm didn't think about it, sure that looks better honestly. Im thinking of creating a separate package/directory fr this + the queries?